### PR TITLE
Change LoadCIF to parse ICSD spacegroup notation

### LINKED ISSUE #19355: LoadCIF with ICSD files
Mantid's `SpaceGroupFactory` and the Inorganic Crystal Structures Database (ICSD) have different conventions for how different settings (cell or origin choice) of the same spacegroup should be specified. This means that some CIF files generated by ICSD cannot be read by LoadCIF. In particular:

 - For the default settings, Mantid uses the plain spacegroup symbol (e.g. `F d -3 m` or `R 3 m`). ICSD for the default settings, for orthorhombic, tetragonal or cubic spacegroups uses the spacegroup symbol with an `S` suffix, e.g. `F d -3 m S`, and for hexagonal spacegroups with an `H` suffix, e.g. `R 3 m H`. 
 - For the second setting, Mantid uses a `:2` suffix for orthorhombic, tetragonal or cubic spacegroups origin choices (e.g. `F d -3 m :2`) whilst ICSD uses a `Z` suffix (e.g. `F d -3 m Z`); for hexagonal spacegroups, in the rhombohedral setting, Mantid uses a `:r` suffix (e.g. `R 3 m :r`) whilst ICSD uses `R` (e.g. `R 3 m R`)

It would thus be good for `LoadCIF` to try to detect the ICSD syntax and translate this into Mantid syntax.

### Expected behavior

`LoadCIF` is able to read an ICSD CIF file and construct a `CrystalStructure` and append it to the designated workspace

### Actual behavior

`LoadCIF` fails with a `RuntimeError: Can not create space group from supplied CIF-file.`

### Steps to reproduce the behavior

Download a CIF file from ICSD, which is in an alternative setting, e.g. http://icsd.cds.rsc.org/viscalc/jsp/jsp/downloadFile.action?submittingParameter=Single%20CIF%20File&saveSingleIDNUM=415476&userAssignedFileName=MyBaseFileName_161405 and use LoadCIF to load it into a workspace.

### Platforms affected

All

### PR DESCRIPTION
**Pull request by mducle**
Changes `LoadCIF` python algorithm to translate spacegroup strings from the notation used by the Inorganic Crystal Structures Database (ICSD) to that used by Mantid. Specifically:

- ICSD uses the suffix `H` and `R` (upper case) to indicate whether the structure uses the hexagonal or rhombohedral settings, whilst Mantid uses `:h` or `:r`
- ICSD uses the suffix `S` for the default origin choice of orthorhombic, tetragonal or cubic spacegroups, and `Z` for the alternate choice, whereas Mantid uses `:1` and `:2`.
- In addition, the default origin choice or setting (`:h` or `:1`) is omitted in the Mantid spacegroup string, but is specified explicitly by ICSD.

**To test:**

Create a workspace and use LoadCIF with a CIF file from ICSD ( http://icsd.cds.rsc.org/search/basic.xhtml ) from spacegroups with multiple settings / origin choices (e.g. any structure with spacegroup numbers 146, 148, 155, 160, 161, 167 for hexagonal / rhombohedral; and 48, 50, 59, 85-88, 125-126, 129-130, 133-134, 201, 203, 222, 227-228).

Before this PR it should fail with a `RuntimeError: Can not create space group from supplied CIF-file.` After this, it should work.

Fixes #19355

*Does not need to be in the release notes.*

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####

- [ ] Is the code of an acceptable quality?
- [ ] Does the code conform to the [coding standards](http://www.mantidproject.org/Coding_Standards)? Is it well structured with small focussed classes/methods/functions?
- [ ] Are there unit/system tests in place? Are the unit tests small and test the a class in isolation?
- [ ] If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- [ ] Do changes function as described? Add comments below that describe the tests performed?
- [ ] How do the changes handle unexpected situations, e.g. bad input?
- [ ] Has the relevant documentation been added/updated?
- [ ] Is user-facing documentation written in a user-friendly manner?
- [ ] Has developer documentation been updated if required?

- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/PythonInterface/plugins/algorithms/LoadCIF.py [hunk 1]
```diff
@@ -76,8 +76,11 @@ def _getSpaceGroupFromString(self, cifData):
 
     def _getCleanSpaceGroupSymbol(self, rawSpaceGroupSymbol):
         # Remove :1 and :H from the symbol. Those are not required at the moment because they are the default.
-        removalRe = re.compile(':[1H]', re.IGNORECASE)
-        return re.sub(removalRe, '', rawSpaceGroupSymbol).strip()
+        # Also substitute 'R' and 'Z' endings used by ICSD to indicate alternative origin choice or settings
+        mappings = {':[1H]':'', ':[1h]':'', ' S$':'', ' H$':'', ' Z$':' :2', ' R$':' :r'}
```
Comment (MichaelWedel) [inline Framework/PythonInterface/plugins/algorithms/LoadCIF.py]: I think the first two could be merged into `':[1Hh]': ''`, but this is nothing major...just one regex less.

**Review by MichaelWedel — COMMENTED** [review]
Code changes look good, I'll test it properly tomorrow.

**Comment by martyngigg** [comment]
It looks like this has been left for a while. @MichaelWedel Will you have chance to test it?

File: Framework/PythonInterface/plugins/algorithms/LoadCIF.py (hunk 1 above)
Comment (mducle) [inline Framework/PythonInterface/plugins/algorithms/LoadCIF.py]: Sorry forgot to push this update...

**Comment by MichaelWedel** [comment]
I don't have access to ICSD at the moment, but I don't see why it wouldn't work the way it's advertised. If anybody with access to ICSD wants to do a test it would be a further insurance.

**Review by MichaelWedel — APPROVED** [review]
Code looks good, tests have been added. If somebody with access to ICSD can confirm that the CIFs are loaded this can be shipped.

**Comment by mducle** [comment]
@MichaelWedel - there's a Korean mirror of the ICSD dataset from about 10 years ago which is not paywalled: http://icsd.kisti.re.kr/first.jsp 
@martyngigg The UK ICSD mirror is: http://icsd.cds.rsc.org/search/basic.xhtml and any STFC ip address should be accepted.

**Comment by martyngigg** [comment]
Looks good to me. I tried out a few files from ICSD and they were loaded successfully.

**Pull request merged**